index.js

```diff
@@ -129,14 +132,42 @@ class ProtomuxRpcClient extends ReadyResource {
     socket.once('close', () => this.connect().catch(safetyCatch))
   }
 
-  async makeRequest (methodName, args, { requestEncoding, responseEncoding }) {
-    return await this._makeRequest(methodName, args, { requestEncoding, responseEncoding })
+  async makeRequest (methodName, args, { requestEncoding, responseEncoding, timeout } = {}) {
+    return await this._makeRequest(methodName, args, { requestEncoding, responseEncoding, timeout })
   }
 
   // Deprecated, just use makeRequest in the next major
   // (no point in having this private)
-  async _makeRequest (methodName, args, { requestEncoding, responseEncoding }) {
-    if (this.opened === false) await this.opening
+  async _makeRequest (methodName, args, { requestEncoding, responseEncoding, timeout } = {}) {
+    timeout = timeout || this.requestTimeout
+
+    // DEVNOTE: there is no need to track timers at object level (to clear them on close):
+    // closing causes the RPC clients to close, causing the request to reject
+    // which triggers the finally that clears the timeout
+    const { resolve, reject, promise } = rrp()
+    const timer = setTimeout(
+      () => { reject(Errors.REQUEST_TIMEOUT()) },
+      timeout
+    )
+    promise.catch(safetyCatch) // no unhandleds
+
+    const reqProm = this._connectAndSendRequest(
+      methodName,
+      args,
+      { requestEncoding, responseEncoding, rpcTimeout: timeout } // Pass on the same timeout, so the RPC request does not stay pending forever after our timeout triggered
```

Comment:
Any reason for the option being named `rpcTimeout`?

Reply:
To make clear that it only applies to the RPC, not to the function as a whole. I'm fine with renaming it to timeout if it seems clearer